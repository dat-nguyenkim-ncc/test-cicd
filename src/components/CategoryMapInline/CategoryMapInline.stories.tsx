import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CategoryMapInline, { CategoryMapInlineProps } from './CategoryMapInline'

export default {
  title: 'Taxonomy/CategoryMapInline',
  component: CategoryMapInline,
} as Meta

const Template: Story<CategoryMapInlineProps> = args => <CategoryMapInline {...args} />

export const Default = Template.bind({})
Default.args = {
  category: { label: 'Category 1', list: ['Subcategory 1'] },
} as CategoryMapInlineProps
