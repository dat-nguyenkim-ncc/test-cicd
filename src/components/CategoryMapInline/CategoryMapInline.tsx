import React from 'react'
import { Flex, Text } from 'theme-ui'
import { MapCategory, ViewInterface } from '../../types'
import { Paragraph } from '../primitives'

export type CategoryMapInlineProps = ViewInterface<{
  category?: MapCategory
}>

const CategoryMapInline = ({ category, sx }: CategoryMapInlineProps) => {
  return (
    <Flex sx={{ alignItems: 'center', ...sx }}>
      <Paragraph bold>{category ? category.label : 'Uncategorised'}</Paragraph>
      {category && (
        <>
          <Text sx={{ mt: '3px', ml: 2, mr: 3, fontSize: '12px' }}>â–¶</Text>
          <Paragraph>{category.list[0]}</Paragraph>
        </>
      )}
    </Flex>
  )
}

export default CategoryMapInline
