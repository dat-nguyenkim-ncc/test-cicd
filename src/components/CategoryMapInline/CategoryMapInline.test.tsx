import * as React from 'react'
import { shallow } from 'enzyme'
import CategoryMapInline from './CategoryMapInline'

describe('CategoryMapInline', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <CategoryMapInline category={{ label: 'Category 1', list: ['Subcategory 1'] }} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
