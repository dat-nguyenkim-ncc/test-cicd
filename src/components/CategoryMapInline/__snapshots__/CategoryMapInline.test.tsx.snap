// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryMapInline matches snapshot 1`] = `ShallowWrapper {}`;
